Neos.Neos/Classes/Routing/NodeIdentityConverterAspect.php

```diff
@@ -34,13 +34,13 @@ class NodeIdentityConverterAspect
      * @param JoinPointInterface $joinPoint the joinpoint
      * @return string|array the context path to be used for routing
      */
-    public function convertNodeToContextPathForRouting(JoinPointInterface $joinPoint)
+    public function convertNodeToContextPathForRouting(JoinPointInterface $joinPoint): array
     {
         $objectArgument = $joinPoint->getMethodArgument('object');
         if ($objectArgument instanceof NodeInterface) {
-            return $objectArgument->getContextPath();
-        } else {
-            return $joinPoint->getAdviceChain()->proceed($joinPoint);
+            return ['__contextNodePath' => $objectArgument->getContextPath()];
```

Comment:
..I mean: Why do we change the aspect from string to array at all?

Reply:
Because that methods was never be supposed to return a string. Also, follow the breadcrumbs, please:

> The method convertObjectToIdentityArray is AOPed in NodeIdentityConverterAspect and is returning a string when a Node is given. This breaks now due to typing added with neos/flow-development-collection#2231

Reply:
See, it's convertObjectToIdentity**Array**() for a reason.

Reply:
OK, I'm trying to follow breadcrumbs. There are 145 of them in my inbox right now :)